design/animation.md

```diff
@@ -0,0 +1,360 @@
+		for (const [k,v] of children) {
+			this[k] = v;
+			v.parent = this;
+		}
+	}
+	public toJSON() {
+		// convert tree into string
+	}
+}
+function AnimateActor(name: string) {
+	return new TargetPath<never>(`actor.${name}`, {
+		transform: new TargetPath<never>("transform", {
+			local: new TargetPath<never>("local", {
+				position: new TargetPath<Vector3>("position", {
+					x: new TargetPath<number>("x"),
+					y: new TargetPath<number>("y"),
+					z: new TargetPath<number>("z")
+				})
+				// also rotation, scale
+			}),
+			app: new TargetPath<never>("app", {
+				/** lookAt replacement */
+				orientToward: new TargetPath<never>("orientToward", {
+					xy: new TargetPath<Vector3>("xy"),
+					y: new TargetPath<Vector3>("y")
+				})
+				// also position, rotation
+			})
+		})
+		// plus most other properties of actor
+	});
+}
+
+const track = {
+	target: AnimateActor("target").transform.local.position
+	// ...
+} as Track<Vector3>;
+```
+
+### References
+
+In addition to transitioning to a constant value on a property, you can also transition to a value on another object.
+This value is re-evaluated every frame on the client, so matches exactly.
+
+```ts
+assets.createAnimation({
+	targets: {
+		"start": "actor"
+		"end": "actor"
+	},
+	tracks: [{
+		target: AnimateActor("start").transform.local.position,
+		keyframes: [{
+			time: 1,
+			value: AnimateActor("end").transform.local.position
+		}]
+	}]
+});
+```
+
+### Instances
+
+Animation instances are animations that have been bound to a particular set of actors and materials. These instances
+can be created by explicitly binding an animation (`Animation.bind(...objects)`), or may be returned during actor
+creation from `Actor.CreateFromPrefab`/`Gltf`/`Library`. Unlike the asset, instances can be played, paused, set speed,
+etc. Animation instances can either be stored directly, or obtained from any of the actors/materials that participate
+in the animation. Instances participate in the patching system, so changing properties on the animation instance
+are pushed down to clients automatically.
+
+An animation instance can be set to start another animation on the instant that it finishes, by setting an instance's
+`continueWith` property. Because it's set in advance, there is zero downtime between one ending and another starting,
+and does not require a round trip.
+
+```ts
+class Animation {
+	bind(...args: Animatible[]): AnimationInstance;
+}
+
+class AnimationInstance {
+	/** Generated unique ID */
+	id: Readonly<string>;
+
+	/** Playback speed multiplier */
+	speed: number;
+	/** What happens when the animation hits the last frame */
+	wrapMode: WrapMode;
+	/** When multiple animations play together, this is the relative strength of this instance */
+	weight: number;
+	/** The current playback time, based on start time and speed */
+	time: number;
+	/** The list of actors/materials that participate in this instance */
+	targets: Readonly<Animatible[]>;
```

Comment:
Is this the map of the instance bindings? Probably needs to be `Readonly<Map<string, Animatable>>`